paimon-core/src/main/java/org/apache/paimon/table/source/InnerStreamTableScanImpl.java

```diff
@@ -194,4 +195,18 @@ public Long checkpoint() {
     public void restore(@Nullable Long nextSnapshotId) {
         this.nextSnapshotId = nextSnapshotId;
     }
+
+    @Override
+    public void notifyCheckpointComplete(@Nullable Long nextSnapshot) {
+        if (nextSnapshot == null) {
+            return;
+        }
+
+        String consumerId = options.consumerId();
+        if (consumerId != null) {
+            snapshotSplitReader
+                    .consumerManager()
+                    .recordConsumer(consumerId, new Consumer(nextSnapshot));
```

Comment:
If two flink programs with the same consumer-id exist simultaneously, what semantics do we guarantee? or  it is necessary we try to  guarantee? cause there is no lock on the file,and anyone with the consumer-id can change it.

Reply:
Good point!
I'm not sure what kind of behavior it should be, so I'll let them each play it out here, not sure who will record success.